### PR TITLE
extra_sans option added to load balancer

### PR DESCRIPTION
**Pull request by hyperbolic2346**
Also cleaned up kubernetes-master charm to use the new method of determining a certificate has changed.

**What this PR does / why we need it**:
Adds an option for the load balancer charm to add extra SAN entries to the generated certificate used by nginx.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Added extra_sans config option to kubeapi-load-balancer charm. This allows the user to specify extra SAN entries on the certificate generated for the load balancer.
```

### CODE REVIEW

File: cluster/juju/layers/kubeapi-load-balancer/reactive/load_balancer.py [hunk 1]
```diff
@@ -44,12 +46,36 @@ def request_server_certificates(tls):
         hookenv.unit_private_ip(),
         socket.gethostname(),
     ]
+    # maybe they have extra names they want as SANs
+    extra_sans = hookenv.config('extra_sans')
+    if extra_sans and not extra_sans == "":
+      sans.extend(extra_sans.split())
```
Comment (Cynerva) [inline cluster/juju/layers/kubeapi-load-balancer/reactive/load_balancer.py]: Indent looks off. Can you make sure this file passes flake8?

File: cluster/juju/layers/kubeapi-load-balancer/reactive/load_balancer.py [hunk 2]
```diff
@@ -44,12 +46,36 @@ def request_server_certificates(tls):
...
+@when('certificates.server.cert.available',
+      'nginx.available', 'tls_client.server.certificate.written')
+def kick_nginx(tls):
+    # we are just going to sighup it, but still want to avoid kicking it
+    # without need
+    if data_changed('cert', tls.get_server_cert()):
+        # certificate changed, so sighup nginx
+        hookenv.log("Certificate information changed, sending SIGHUP to nginx")
+        host.service_restart('nginx')
+    remove_state('tls_client.server.certificate.written')
```
Comment (ktsakalozos) [inline cluster/juju/layers/kubeapi-load-balancer/reactive/load_balancer.py]: This is not polite :) . We remove a state from another layer, shouldn't we be asking the other layer to do the removal for us?

**Comment by hyperbolic2346** [comment]
This now depends on https://github.com/juju-solutions/layer-tls-client/pull/10

**Comment by Cynerva** [comment]
/lgtm

layer-tls-client dependency is merged, and this looks good to me. Thanks @hyperbolic2346

**Pull request merged**